recipes/iwyu/build.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+mkdir build
+cd build
+
+cmake \
+  -G "Unix Makefiles" \
+  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+  -DCMAKE_PREFIX_PATH=${PREFIX} \
+  -DCMAKE_BUILD_TYPE=release \
+  -DCMAKE_C_COMPILER=${CC} \
+  -DCMAKE_CXX_COMPILER=${CXX} \
+  ..
+
+cmake --build .
+cmake --install . -v
```

Comment:
Is there a ctest or make check we can run here?

Reply:
good catch @ocefpaf ! it seems that ctest is available and it seems that some tests are failing. I am investigating that now. thank you so much! really appreciate that :)

Reply:
it is raising weird issue about the standard library access https://github.com/include-what-you-use/include-what-you-use/issues/1033#issuecomment-1107884014

any idea about that?

Reply:
I'm not sure I follow the upstream comment, do you need a c++ compiler in the test environment?

Reply:
thanks @ocefpaf for your attention here :)

this is part of the log from ctest:

```
14/143 Testing: cxx.test_badinc
14/143 Test: cxx.test_badinc
Command: "/mnt/sda1/storage/miniconda3/envs/conda-build/conda-bld/include-what-you-use_1650805645046/_build_env/bin/python" "run_iwyu_tests.py" "--run-test-file=tests/cxx/badinc.cc" "--" "/mnt/sda1/storage/miniconda3/envs/conda-build/conda-bld/include-what-you-use_1650805645046/work/build/bin/include-what-you-use"
Directory: /mnt/sda1/storage/miniconda3/envs/conda-build/conda-bld/include-what-you-use_1650805645046/work
"cxx.test_badinc" start time: Apr 24 09:13 -04
Output:
----------------------------------------------------------
INFO:root:Testing iwyu on tests/cxx/badinc.cc
>>> Running /mnt/sda1/storage/miniconda3/envs/conda-build/conda-bld/include-what-you-use_1650805645046/work/build/bin/include-what-you-use -Xiwyu --verbose=3 -Xiwyu --mapping_file=tests/cxx/badinc.imp -I .  tests/cxx/badinc.cc
tests/cxx/badinc.cc:60:10: fatal error: 'typeinfo' file not found
#include <typeinfo>   // for typeid
         ^~~~~~~~~~
```

I am also new to iwyu ... so maybe it is also a lack of understanding about how to test it XD
but it seems that it depends on the compiler for running as well

Reply:
mmm maybe I would need to patch the tests: https://github.com/include-what-you-use/include-what-you-use/blob/master/iwyu_tool_test.py#L103
or maybe I could just skip these "problematic" tests ..
what is the CF common approach for that?

Reply:
If you can make a case that those test do not matter you can skip them for now, get this merged, and open an issue to track them in the feedstock.

Reply:
You can use `ctest --output-on-failure` to show why the tests fail, and you can also use `-R` to filter the failing tests with regular expressions.

https://cmake.org/cmake/help/latest/manual/ctest.1.html

Reply:
@ocefpaf I am checking with the iwyu community about this problem ... I am not sure yet if it is a big deal or not

Reply:
I isolated one test and ran it with more flags, and it worked fine:

```
export iwyu=/mnt/sda1/storage/miniconda3/envs/conda-build/conda-bld/include-what-you-use_1651536695168/work/build/bin/include-what-you-use

$iwyu -Xiwyu --verbose=3 \
  -I${PREFIX}/include \
  -I${BUILD_PREFIX}/x86_64-conda-linux-gnu/include/c++/11.2.0/ \
  -I${BUILD_PREFIX}/x86_64-conda-linux-gnu/include/c++/11.2.0/x86_64-conda-linux-gnu/ \
  -I . tests/cxx/uses_printf.cc
  ```

so I guess that it is safe to skip the test that are failing for now, and try to fix that in the feedstock